src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -968,28 +972,31 @@ export class GuidedTourService {
      * @return guided tour step with defined orientation
      */
     private setTourOrientation(step: TourStep): TourStep {
-        const convertedStep = cloneDeep(step);
-        if (convertedStep.orientation && !(typeof convertedStep.orientation === 'string') && (convertedStep.orientation as OrientationConfiguration[]).length) {
-            (convertedStep.orientation as OrientationConfiguration[]).sort((a: OrientationConfiguration, b: OrientationConfiguration) => {
-                if (!b.maximumSize) {
-                    return 1;
-                }
-                if (!a.maximumSize) {
-                    return -1;
-                }
-                return b.maximumSize - a.maximumSize;
-            });
+        if (step) {
```

Comment:
Style nit: Prefer returning early over nesting (that also reduces the diff)
```
if (!step) {
    return step;
}

const convertedStep = cloneDeep(step);
[...]
```

Reply:
will do that!